DEVX-4572 Add Android push notification upload tool docs (#3618)
Nexmo/nexmo-developer: _documentation/en/client-sdk/setup/set-up-push-notifications/android.md

> **NOTE:** In order to apply any methods on the `NexmoClient` object (for example answer a call, hangup, and so on), the `NexmoClient` has to be initialized and the user has to be [logged in](/client-sdk/getting-started/add-sdk-to-your-app/android) to it.

> ## Link Vonage backend push service to Firebase

Reviewer: I believe the original title worked better - we're only affecting change to one application and not the entore Vonage system. I would also not call it a backend - will be confusion for readers.

To connect Vonage backend push service with Firebase you will need the following:

1. Vonage API Application id
3. Firebase project id
4. Firebase token

### Get Vonage application Id

> Obtain your Vonage API Application id. You can access the existing application in the [dashboard](https://dashboard.nexmo.com/voice/your-applications). If you don't have an application already you can create the new application via [Nexmo CLI](/client-sdk/setup/create-your-application).

Reviewer: existing application => existing application**s**

### Get Firebase project Id

Get your Firebase project id from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console -> Project settings -> General`.

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-id.png)

### Get Firebase token

> Get your Firebase token from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key. 

Reviewer: Are we using the private key as a token? Readers might find confusing. If the answer is yes, consider renaming `FIREBASE_TOKEN` to `FIREBASE_PRIVATE_KEY`

Author: It looks like Firebase nomenclature is calling this a `Server key` so renaming to `FIREBASE_SERVER_KEY` will be more accurate (same as a screenshot)

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-token.png)

> ## Link the Vonage backend push service with the Firebase application

Reviewer: Same comment as before - Let's use: "Connect your Vonage application to Firebase"

You link Vonage backend push service with the Firebase application by making a POST request. You can to this request using Upload Tool or making a POST request directly.
### Using the Upload Tool

The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), does so with a user interface.

> Once you have the tool running, enter your Vonage Application ID, private key file (located in nexmo application folder), Firebase project id, Firebase token and click upload. The status of your upload will be shown on the page once it is complete:

Reviewer: Also, I don't believe we mentioned any application folder so we should remove the mention of it.

3. Firebase project id
4. Firebase token

### Get Vonage application Id

Obtain your Vonage API Application id. You can access the existing applications in the [dashboard](https://dashboard.nexmo.com/voice/your-applications). If you don't have an application already you can create the new application via [Nexmo CLI](/client-sdk/setup/create-your-application).

### Get Firebase project Id

Get your Firebase project id from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console -> Project settings -> General`.

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-id.png)

### Get Firebase Server Key

Get your Firebase Server Key from the [Firebase console](https://console.firebase.google.com/). Navigate to `Firebase console ->  Project settings -> Service accounts` and generate a new private key. 

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-project-settings.png)

![](/screenshots/setup/client-sdk/set-up-push-notifications/firebase-token.png)

## Connect Vonage API application to Firebase

> You connect Vonage backend push service with the Firebase application by making a POST request. You can to this request using Upload Tool or making a POST request directly.

Reviewer 2: ```suggestion
You connect Vonage backend push service with the Firebase application by making a POST request. You can to this request using the Upload Tool or making a POST request directly.
```

### Using the Upload Tool

> The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), does so with a user interface.

Reviewer 2: ```suggestion
The Android Push Certificate Uploading Tool, available on [GitHub](https://github.com/nexmo-community/android-push-uploader), allows you to upload with a user interface.
```

   https://api.nexmo.com/v1/applications/$VONAGE_APP_ID/push_tokens/android  
```

> If all the values are correct you should see `200` response code in the terminal.

Reviewer: Not entirely correct - there is no validation at this point. 200 means that Vonage got your data and stored it but haven't actually checked that they are working. Worth making that clearer.
